horizomer/utils/tree.py

```diff
@@ -44,6 +44,31 @@ def compare_topology(tree1, tree2):
     return n2p1 == n2p2
 
 
+def intersect_trees(tree1, tree2):
+    """Shrink two trees to contain only overlapping taxa.
+
+    Parameters
+    ----------
+    tree1 : skbio.TreeNode
+        first tree to intersect
+    tree2 : skbio.TreeNode
+        second tree to intersect
+
+    Returns
+    -------
+    tuple of two TreeNodes
+        resulting trees containing only overlapping taxa
+    """
+    taxa1 = set([tip.name for tip in tree1.tips()])
+    taxa2 = set([tip.name for tip in tree2.tips()])
+    taxa_lap = taxa1.intersection(taxa2)
+    if len(taxa_lap) == 0:
```

Comment:
logical understanding: what if tree1 has one tip with name 'a' and tree2 two tips (but at different sub-trees) with name 'a'?

Reply:
This will break the function. However, I plan to used another function to check if a tree have unique taxon names prior to this function.

Reply:
You might want to add a comment about this in the docstring then

Reply:
Let me just add a function to test uniqueness.